crates/masking/src/secret.rs

```diff
@@ -152,3 +152,14 @@ where
         SecretValue::default().into()
     }
 }
+
+impl<SecretValue, MaskingStrategy, T> AsRef<T> for Secret<SecretValue, MaskingStrategy>
+where
+    SecretValue: AsRef<T>,
+    MaskingStrategy: Strategy<SecretValue>,
+    T: ?Sized,
+{
+    fn as_ref(&self) -> &T {
+        self.inner_secret.as_ref()
+    }
+}
```

Comment:
Please avoid adding this implementation: we'd prefer knowing where the inner value of the secret is being accessed by `peek()` / `expose()` as compared to implicit conversions such as `Deref` or `AsRef`.